src/exercism/ResistorColor.java

```diff
@@ -0,0 +1,25 @@
+class ResistorColor {    
+    String[] colorNames = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};
+
+    int colorCode(String color) {
+        int code = 0;  
```

Comment:
it is quite dangerous to set code to 0. In case somebody passes a colour that does not exist in colorNames, then your code will return 0, which would suggest colorNames[0], which is not true.

A better approach would be to initialize `int code = -1`, and then after the for loop throw an exception if the colour was not found